pkg/templates/kubeadm/v1beta1/kubeadm.go

```diff
@@ -35,9 +35,19 @@ import (
 func NewConfig(ctx *util.Context, host kubeoneapi.HostConfig) ([]runtime.Object, error) {
 	cluster := ctx.Cluster
 
+	nodeIP := host.PrivateAddress
+	if nodeIP == "" {
+		nodeIP = host.PublicAddress
+	}
+
 	nodeRegistration := kubeadmv1beta1.NodeRegistrationOptions{
-		Name:             host.Hostname,
-		KubeletExtraArgs: map[string]string{"node-ip": host.PrivateAddress},
+		Name: host.Hostname,
+		KubeletExtraArgs: map[string]string{
+			"anonymous-auth":      "false",
```

Comment:
What are those new arguments doing and why we need them?

Reply:
They match how machine-controller init worker nodes. Also slightly increate security.